test/ServerRequestFactoryTest.php

```diff
@@ -570,4 +570,15 @@ public function testMarshalRequestUriPrefersRequestUriServerParamWhenXOriginalUr
         $uri = marshalUriFromSapi($server, $headers);
         $this->assertSame('/requested/path', $uri->getPath());
     }
+
+    public function testServerRequestFactoryHasAWritableEmptyBody()
+    {
+        $factory = new ServerRequestFactory();
+        $request = $factory->createServerRequest('GET', '/');
+        $body = $request->getBody();
+
+        $this->assertTrue($body->isWritable());
```

Comment:
Hmm, I don't think the global request body should be writable here

Reply:
That's not the "global request body" but a body of a server request just created without global values. To me, creating an empty `php://temp` body that cannot be writable makes no sense.

Reply:
This is the request coming in from global state: if anybody is trying to write to this body, something went extremely wrong IMO.

Reply:
What?! `createServerRequest` should create a server request using only the values passed in the constructor (method, uri and serverParams), and never use global variables. There's the `fromGlobals` to do that. Or I'm missing something?

An example of use case [is testing](https://github.com/middlewares/payload/blob/master/tests/PayloadTest.php#L33).

Reply:
Ah yes, I got confused here, my mistake!